src/test/scala/com/microsoft/hyperspace/index/plananalysis/ExplainTest.scala

```diff
@@ -436,6 +436,178 @@ class ExplainTest extends SparkFunSuite with HyperspaceSuite {
+      .append(defaultDisplayMode.newLine)
+      .append(defaultDisplayMode.newLine)
+      .append("=============================================================")
+      .append(defaultDisplayMode.newLine)
+      .append("Indexes used:")
+      .append(defaultDisplayMode.newLine)
+      .append("=============================================================")
+      .append(defaultDisplayMode.newLine)
+      .append(s"joinIndex:$joinIndexPath")
+      .append(defaultDisplayMode.newLine)
+      .append(defaultDisplayMode.newLine)
+      .append("=============================================================")
+      .append(defaultDisplayMode.newLine)
+      .append("Physical operator stats:")
+      .append(defaultDisplayMode.newLine)
+      .append("=============================================================")
+      .append(defaultDisplayMode.newLine)
+      .append("+------------------+-------------------+------------------+----------+")
+      .append(defaultDisplayMode.newLine)
+      .append("| Physical Operator|Hyperspace Disabled|Hyperspace Enabled|Difference|")
+      .append(defaultDisplayMode.newLine)
+      .append("+------------------+-------------------+------------------+----------+")
+      .append(defaultDisplayMode.newLine)
+      .append("|           *Filter|                  1|                 2|         1|")
+      .append(defaultDisplayMode.newLine)
+      .append("|     *InputAdapter|                  4|                 2|        -2|")
+      .append(defaultDisplayMode.newLine)
+      .append("|          *Project|                  1|                 2|         1|")
+      .append(defaultDisplayMode.newLine)
+      .append("|   *ReusedExchange|                  1|                 0|        -1|")
+      .append(defaultDisplayMode.newLine)
+      .append("|     *Scan parquet|                  1|                 2|         1|")
+      .append(defaultDisplayMode.newLine)
+      .append("|  *ShuffleExchange|                  1|                 0|        -1|")
+      .append(defaultDisplayMode.newLine)
+      .append("|             *Sort|                  2|                 0|        -2|")
+      .append(defaultDisplayMode.newLine)
+      .append("|*WholeStageCodegen|                  4|                 3|        -1|")
+      .append(defaultDisplayMode.newLine)
+      .append("|     SortMergeJoin|                  1|                 1|         0|")
+      .append(defaultDisplayMode.newLine)
+      .append("+------------------+-------------------+------------------+----------+")
+      .append(defaultDisplayMode.newLine)
+      .append(defaultDisplayMode.newLine)
+    // scalastyle:on filelinelengthchecker
+
+    val selfJoinDf = df.join(df, df("Col1") === df("Col1"))
+
+    // Materialize the lazily evaluated optimized plan before plan comparison.
+    spark.enableHyperspace()
+    selfJoinDf.queryExecution.optimizedPlan
+    spark.disableHyperspace()
```

Comment:
Note: the test is same as test titled "Testing default display mode" except for these three lines which materialize the optimized plan.
This materialization fails the test with the original code